plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -493,61 +540,84 @@ protected function initialize()
-                            $filters = array_merge($filters, ['diacritics_lowercase']);
-                        }
-
-                        $this->config['index']['configuration']['analysis']['analyzer'][$key]['char_filter'] = $filters;
-                    }
-                }
-
-                $this->index->create(
-                    $this->config['index']['configuration'],
-                    ['recreate' => true]
-                );
-            }
+        // Load and normalize mappings
+        $this->loadAndNormalizeMappings();
 
-            // Load and normalize mappings
-            $this->loadAndNormalizeMappings();
+        // Iterate over types (actor, informationobject, ...)
+        foreach ($this->mappings as $indexName => $indexProperties) {
+            $indexName = 'Qubit'.sfInflector::camelize($indexName);
+            $prefixedIndexName = $this->config['index']['name'].'_'.strtolower($indexName);
+            $index = $this->client->getIndex($prefixedIndexName);
+            $this->index->addIndex($indexName, $index);
 
-            // Iterate over types (actor, informationobject, ...)
-            foreach ($this->mappings as $typeName => $typeProperties) {
-                $typeName = 'Qubit'.sfInflector::camelize($typeName);
+            try {
+                $index->open();
+            } catch (Exception $e) {
+                // If the index has not been initialized, create it
+                if ($e instanceof \Elastica\Exception\ResponseException) {
+                    $this->configureFilters();
+
+                    // In ES 7.x if the mapping type is updated to a dummy type,
+                    // this may need to include a param for include_type_name
+                    // set to false in order to avoid automatically creating a
+                    // type for the index that was just created
+                    $index->create(
+                        $this->config['index']['configuration'],
+                        ['recreate' => true]
+                    );
+                }
 
                 // Define mapping in elasticsearch
                 $mapping = new \Elastica\Type\Mapping();
-                $mapping->setType($this->index->getType($typeName));
-                $mapping->setProperties($typeProperties['properties']);
+
+                // Setting a dummy type since it is required in ES 6.x
+                // but it can be removed in 7.x when it becomes optional
+                $mapping->setType($index->getType(self::ES_TYPE));
+                $mapping->setProperties($indexProperties['properties']);
 
                 // Parse other parameters
-                unset($typeProperties['properties']);
-                foreach ($typeProperties as $key => $value) {
+                unset($this->mapping[$indexName]->indexProperties['properties']);
+                foreach ($this->mapping[$indexName]->indexProperties as $key => $value) {
```

Comment:
Can't you do this directly over the `$indexProperties` variable? Why accessing the mapping again?